pyaedt/hfss.py

```diff
@@ -2478,13 +2480,35 @@ def assign_primary(self, face, u_start, u_end, reverse_v=False, coord_name="Glob
         return self._create_boundary(primary_name, props, "Primary")
 
     def _create_pec_cap(self, sheet_name, obj_name, pecthick):
-        # TODO check method
-        obj = self.modeler[sheet_name].clone()
+        """Create a PEC object to back a wave port.
+
+
+        Parameters
+        ----------
+        sheet_name : str
+            Name of the 3d object touching the port surface.
+        obj_name : ???
+        pecthick : float
+            Thickness of the PEC cap
+
+        Returns
+        -------
+
+        """
+        if isinstance(sheet_name, str) and isinstance(obj_name, cad.elements3d.FacePrimitive):
+            obj = obj_name.create_object()  # Create face object of type cad.object3d.Object3d from FacePrimitive
+            oname = obj_name._object3d.name
```

Comment:
@dcrawforAtAnsys obj_name already is a FacePrimitve object, correct? so you can already access name prop